pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -131,6 +131,11 @@ public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService e
     QueryContext queryContext = queryRequest.getQueryContext();
     LOGGER.debug("Incoming request Id: {}, query: {}", requestId, queryContext);
 
+    boolean enableTrace = queryRequest.isEnableTrace();
+    if (enableTrace) {
+      Tracing.getTracer().register(requestId);
```

Comment:
(MAJOR) Register trace here can cause leak because there is no guarantee trace is unregistered

Reply:
Yes, this is problematic for the built in implementation. I will ensure it is unregistered.

Reply:
fixed